packages/devtools-evm-hardhat/src/transactions/signer.ts

```diff
@@ -1,16 +1,22 @@
 import pMemoize from 'p-memoize'
 import type { OmniSignerFactory } from '@layerzerolabs/devtools'
-import { OmniSignerEVM } from '@layerzerolabs/devtools-evm'
+import { GnosisOmniSignerEVM, OmniSignerEVM } from '@layerzerolabs/devtools-evm'
 import { createProviderFactory } from '@/provider'
+import { createGetHreByEid } from '@/runtime'
 
 export const createSignerFactory = (
     addressOrIndex?: string | number,
-    providerFactory = createProviderFactory()
-): OmniSignerFactory<OmniSignerEVM> => {
+    providerFactory = createProviderFactory(),
+    networkEnvironmentFactory = createGetHreByEid()
+): OmniSignerFactory => {
     return pMemoize(async (eid) => {
         const provider = await providerFactory(eid)
         const signer = provider.getSigner(addressOrIndex)
 
-        return new OmniSignerEVM(eid, signer)
+        const env = await networkEnvironmentFactory(eid)
+        const safeConfig = env.network.config.safeConfig
+        return safeConfig
+            ? new GnosisOmniSignerEVM(eid, signer, safeConfig.safeUrl, safeConfig)
+            : new OmniSignerEVM(eid, signer)
```

Comment:
Okay for now but in general this will not allow people to directly influence the output of this function (since the decision is made internally based on the hardhat config). I would prefer a more explicit way of doing this:

- Create a new factory for the safe signer of type `OmniSignerFactory<GnosisOmniSignerEVM>`
- Then you can, e.g. based on some CLI flag, pass this or that factory to `subtask.sign-and-send.ts`

This way the behavior can be controlled directly by the consumer

Reply:
Let's handle this in a follow-up.